File: src/main/java/com/company/erp/features/chats/dto/CreateConversationRequest.java
package com.company.erp.features.chats.dto;

import com.company.erp.features.chats.entity.ConversationType;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.util.Set;

public record CreateConversationRequest(
        @NotNull ConversationType type,
        @NotEmpty Set<Long> memberIds,
        @Size(max = 255) String name,
        @Size(max = 1024) String avatarUrl
) {
}

File: app/Features/Chat/Requests/CreateConversationRequest.php
<?php

namespace App\Features\Chat\Requests;

use App\Features\Chat\Models\ConversationType;
use App\Support\Http\ApiFormRequest;
use Illuminate\Validation\Rule;

/** Port of the Spring CreateConversationRequest. */
class CreateConversationRequest extends ApiFormRequest
{
    public function rules(): array
    {
        return [
            'type' => ['required', Rule::in(ConversationType::values())],
            'memberIds' => ['required', 'array', 'min:1'],
            'memberIds.*' => ['integer'],
            'name' => ['nullable', 'string', 'max:255'],
            'avatarUrl' => ['nullable', 'string', 'max:1024'],
        ];
    }
}
